src/Compilers/VisualBasic/Portable/Symbols/SynthesizedSymbols/GeneratedNameKind.vb

```diff
@@ -43,6 +46,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Return GeneratedNameKind.StateMachineAwaiterField
             ElseIf name.StartsWith(StringConstants.StateMachineHoistedUserVariablePrefix, StringComparison.Ordinal) Then
                 Return GeneratedNameKind.StateMachineHoistedUserVariableField
+            ElseIf name.StartsWith(AnonymousTypeTemplateNamePrefix, StringComparison.Ordinal) Then
+                Return GeneratedNameKind.AnonymousType
+            ElseIf name.Equals(StringConstants.It, StringComparison.Ordinal) OrElse
+                    name.Equals(StringConstants.It1, StringComparison.Ordinal) OrElse
+                    name.Equals(StringConstants.It2, StringComparison.Ordinal) Then
+                Return GeneratedNameKind.TransparentIdentifier
```

Comment:
Perhaps just name.StartsWith(StringConstants.It, StringComparison.Ordinal), after checking for ItAnonymous.